### PR TITLE
[vcpkg] Harden expand environment strings path with explicit integer overflow checks and resistance to CP_ACP.

### PR DESCRIPTION
**Pull request by BillyONeal**

### CODE REVIEW

File: toolsrc/src/vcpkg/commands.edit.cpp [hunk 1]
```diff
@@ -44,8 +45,39 @@ namespace vcpkg::Commands::Edit
...
         return output;
     }
+
+    std::string expand_environment_strings(const std::string& input)
+    {
+        const auto widened = vcpkg::Strings::to_utf16(input);
+        std::wstring result;
+        result.resize(result.capacity());
+        bool done;
+        do
```
Comment (strega-nil) [inline toolsrc/src/vcpkg/commands.edit.cpp:55]: why do we loop here, as opposed to just expanding once? is that the expected behavior?

Comment (BillyONeal) [reply toolsrc/src/vcpkg/commands.edit.cpp:55]: If the value of an environment variable is changed between the calls, the loop makes it safe. Unlikely sure but means there's no chance of running off the end of the buffer.

Comment (strega-nil) [reply toolsrc/src/vcpkg/commands.edit.cpp:55]: Is that possible? We're not threaded, and I don't know how else one would change the environment variables out from under us.

Comment (BillyONeal) [reply toolsrc/src/vcpkg/commands.edit.cpp:55]: >We're not threaded

Right now we're not, sure, but if we ever changed that TOCTOU bugs suck :)

Comment (strega-nil) [reply toolsrc/src/vcpkg/commands.edit.cpp:55]: I mean fair enough.

**Pull request merged**